opensearch-operator/pkg/reconcilers/cluster.go

```diff
@@ -286,6 +286,71 @@ func (r *ClusterReconciler) reconcileNodeStatefulSet(nodePool opsterv1.NodePool,
 		}
 	}
 
+	// Habdle PDB
+	needToUpdate := false
+	// Check if it needed
+	if nodePool.Pdb.EnablePDB {
+
+		pdb := v1.PodDisruptionBudget{}
+		newpdb := v1.PodDisruptionBudget{}
+		// Check if it already exist
+		if err = r.Get(r.ctx, client.ObjectKey{Name: r.instance.Name + "-" + nodePool.Component + "-pdb", Namespace: r.instance.Namespace}, &pdb); err == nil {
+			// IF the resource exist, start to check if there is any changes in the configurations
+			if pdb.Spec.MaxUnavailable != nil {
+				if pdb.Spec.MaxUnavailable.IntVal != nodePool.Pdb.MaxUnavailable.IntVal {
+					pdb.Spec.MaxUnavailable.IntVal = nodePool.Pdb.MaxUnavailable.IntVal
+					needToUpdate = true
+				}
+			}
+			if pdb.Spec.MinAvailable != nil {
+				if pdb.Spec.MinAvailable.IntVal != nodePool.Pdb.MinAvailable.IntVal {
+					pdb.Spec.MinAvailable.IntVal = nodePool.Pdb.MinAvailable.IntVal
+					needToUpdate = true
+				}
+			}
+			// Update configuration and requeue
+			if needToUpdate {
+				r.logger.Info("Updating PDB ")
+				if err = r.Update(r.ctx, &pdb); err != nil {
+					return result, err
+				}
+			}
+		} else {
+
+			// If it not exists , build && create it but before check that all the parameters are provided
+			if (nodePool.Pdb.MinAvailable != nil && nodePool.Pdb.MaxUnavailable != nil) || (nodePool.Pdb.MinAvailable == nil && nodePool.Pdb.MaxUnavailable == nil) {
+				r.logger.Info(" Aborting PodDisruptionBudget creation - Please provided one parameter (MinAvailable OR MaxUnavailable) in order to create PodDisruptionBudget resource")
+
+			} else {
```

Comment:
I think you should also have logic to generate a default PDB if neither MinAvailable or MaxUnavailable are set. A good default could be to set `MaxUnAvailable=1`.

Reply:
So i thought about that also, but i could not find any docs of ES/OS that they are reocmmend about default value... 
 only that : https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-pod-disruption-budget.html
 and the value here is true.
 I dont what to make customers to expirince things that they are not defined.

Reply:
In the second paragraph of the elastic docs you linked is written "ECK manages a default PDB per Elasticsearch resource. It allows one Elasticsearch Pod to be taken down". That for me is a clear indication that the default is `MaxUnavailable=1`. So doing it the same for our operator would be consistent. 

> I dont what to make customers to expirince things that they are not defined.

They need to explicitly enable the budget anyway, and as long as it is documented what the default budget is I don't see a problem with that. On the contrary, one of the jobs of the operator is to do sensible defaults, so setting a default config for the budget if the user enables it is normal behaviour IMO.